src/input.rs

```diff
@@ -159,7 +159,13 @@ impl Action {
             Action::Paste |
             Action::PasteSelection => {
                 Clipboard::new()
-                    .and_then(|clipboard| clipboard.load_selection())
+                    .and_then(|clipboard|
+                              match *self {
```

Comment:
The redundant match is kind of a bummer. It would be nice if each of the top level match arms did something like

```
Action::Paste => Clipboard::new()
    .and_then(|clipboard| clipboard.load_primary())
    .map(|contents| self.paste(contents)),
```

which would require refactoring the `map` closure into a method on `Action` called `paste`.